### ISSUE TITLE
Illegal characters in filename

### ISSUE THREAD

**Issue by Mr-VIT** [jviotti/electron-json-storage#26]
Need to encode special characters such as `? : ` before using key in file name.

**Comment by jviotti** [comment]
Hi @Mr-VIT ,

Can you provide an example? I've tried the following, which seems to work fine here (OS X, at least):

```js
storage.set('foo?bar:baz', { foo: 'bar' }, callback);
```

**Comment by jviotti** [comment]
I'm sending a PR adding such test case to the test suite, and lets see if the CI servers are able to reproduce.

**Comment by jviotti** [comment]
Okay, I'm able to reproduce an `ENOENT` error in Windows. Not sure if its the same one you're reporting:

```
Error: ENOENT: no such file or directory, open 'C:\Users\appveyor\AppData\Local\Temp\1\electron-mocha-cFwk7A\foo?bar:baz.json'
```

**Comment by Mr-VIT** [comment]
Windows: if `?` `Error: ENOENT: no such file or directory`
if `foo:bar` using NTFS stream (ADS) `bar` in file `foo`

**Comment by jviotti** [comment]
@Mr-VIT I've just sent a PR with a fix. Can you confirm it works for you? You can install it your project directly from GitHub like this:

```sh
npm install --save jviotti/electron-json-storage#special-characters
```

**Comment by Mr-VIT** [comment]
Now fine.

**Comment by jviotti** [comment]
I've published v2.0.1, which includes this fix, to npm.
